qiskit/transpiler/passmanager.py

```diff
@@ -462,6 +472,9 @@ def _replace_error(meth):
     def wrapper(*meth_args, **meth_kwargs):
         try:
             return meth(*meth_args, **meth_kwargs)
+        except TranspilerError:
+            # If it's already a `TranspilerError` subclass, don't erase the extra information.
```

Comment:
I remember discussing this offline, but `TranspilerError` being a subclass of `PassManagerError` seems so weird. But it's hard to change this now so this makes sense.

Reply:
I assume it was intended as a backwards-compatibility shim for a move to `PassManagerError` during the genericisation of `PassManager`, but I don't really know.